src/app/GitUI/LeftPanel/FavoritesTree.cs

```diff
@@ -0,0 +1,168 @@
+﻿using GitCommands;
+using GitCommands.Git;
+using GitCommands.Remotes;
+using GitExtensions.Extensibility;
+using GitExtensions.Extensibility.Git;
+using GitUI.CommandDialogs;
+using GitUI.Properties;
+using GitUI.UserControls.RevisionGrid;
+using Microsoft;
+
+namespace GitUI.LeftPanel
+{
+    internal sealed class FavoritesTree : BaseRefTree
+    {
+        private readonly IAheadBehindDataProvider? _aheadBehindDataProvider;
+        private readonly IRevisionGridInfo _revisionGridInfo;
+        private readonly CachedFavorites _cachedFavorites = new();
+        private readonly IGitUICommandsSource _refUiCommand;
+
+        public FavoritesTree(TreeNode treeNode, IGitUICommandsSource uiCommands, IAheadBehindDataProvider? aheadBehindDataProvider, ICheckRefs refsSource, IRevisionGridInfo revisionGridInfo)
+            : base(treeNode, uiCommands, refsSource, RefsFilter.Remotes | RefsFilter.Heads)
+        {
+            _aheadBehindDataProvider = aheadBehindDataProvider;
+            _revisionGridInfo = revisionGridInfo;
+            _refUiCommand = uiCommands;
+
+            if (refsSource is GitModuleControl { Module.WorkingDirGitDir: not null } gitModuleControl)
+            {
+                _cachedFavorites.Location = gitModuleControl.Module.WorkingDirGitDir;
+            }
+        }
+
+        public void Add(NodeBase node)
+        {
+            if (node is BaseRevisionNode baseRevisionNode && baseRevisionNode.ObjectId != null)
+            {
+                _cachedFavorites.Add(baseRevisionNode.ObjectId.ToString());
+            }
+
+            Refresh(new FilteredGitRefsProvider(UICommands.Module).GetRefs, true);
+        }
+
+        public void Remove(NodeBase node)
+        {
+            if (node is BaseRevisionNode baseRevisionNode && baseRevisionNode.ObjectId != null)
+            {
+                _cachedFavorites.Remove(baseRevisionNode.ObjectId.ToString());
+            }
+
+            Refresh(new FilteredGitRefsProvider(UICommands.Module).GetRefs, true);
+        }
+
+        protected override Nodes FillTree(IReadOnlyList<IGitRef> branches, CancellationToken token)
+        {
+            FavoriteNode remoteNode = new(this, TranslatedStrings.Remotes, nameof(Images.BranchRemoteRoot)) { Visible = true };
+            FavoriteNode localNode = new(this, TranslatedStrings.Local, nameof(Images.BranchLocalRoot)) { Visible = true };
```

Comment:
We should consider changing the text `TranslatedStrings.Local` from "Local" to "Local branches" for the entire left panel.